app/models/ext_management_system.rb

```diff
@@ -22,17 +22,15 @@ def self.leaf_subclasses
   end
 
   def self.supported_subclasses
-    subclasses.flat_map do |s|
-      s.subclasses.empty? ? s : s.supported_subclasses
-    end
```

Comment:
@Fryguy not sure why we had `supported_subclasses` and `leaf_subclasses`...they look like they both do exactly the same thing just in different ways

`leaf_subclasses` uses `descendants` and selects `subclasses.empty?` where `supported_subclasses` recursively calls through the subclasses then filters on `subclasses.empty?`

There also doesn't appear to be anything "supported" about these subclasses other than they are leaf classes.

I'm thinking of turning `supported_subclasses` into the set of leaf_subclasses that are supported by `Vmdb::PermissionStores.instance.supported_ems_type?` but there are existing callers that we need to make sure aren't expecting the full list, supported or not.

The other option is to make supported_subclasses an alias of leaf_subclasses and add a new method (actually_supported_subclasses?) to do this.

Reply:
that's weird...I think it always should have been where leaf_subclasses returned, well, all of the leaf subclasses.  Then, supported subclasses should just filter that set.

Reply:
FWIW, leaf_subclasses is a part of more_core_extensions now https://github.com/ManageIQ/more_core_extensions/blob/master/lib/more_core_extensions/core_ext/class/hierarchy.rb#L44-L51

Reply:
Oh even better!

Reply:
> I'm thinking of turning supported_subclasses into the set of leaf_subclasses that are supported by Vmdb::PermissionStores.instance.supported_ems_type? but there are existing callers that we need to make sure aren't expecting the full list, supported or not.

supported should have always been the filtered set, so I don't think the callers should have expected otherwise.